custom_components/energyscore/sensor.py

```diff
@@ -195,6 +193,14 @@ def process_new_data(self):
             energy_used = False
             _LOGGER.debug("Need more energy data to update %s", self._name)
 
+        entity_reg = er.async_get(self.hass)
```

Comment:
I see that this would work fine, but something in me says that it is wrong to do this for every data update cycle. Do you think it would be bad to put in async_setup_platform function and just wait for the price entity to be ready before actually adding EnergyScore sensor. What do you think?

Reply:
I don't know how to do what you propose: what happens if the entity never appears (misconfiguration for instance)?

Cost of doing this at each iteration (every 10minutes) is really low: it is a simple dictionary access.
I get the simplicity argument though.
I'll try to propose something.

Reply:
A quick test show that:
- entity seems to be created sequentially, and we can't control the order between energyscore sensor and the price sensor
- HA awaits on `async_setup_platform` so we cannot simply "wait" for other sensor to appear. Unless we spawn some kind of future where we wait and then add our newly created sensor. But it goes out of control for many reasons: we have no way to know if platform setup happens correctly, it means our entities will be created beyond HA regular bootstrap.

So technically it is likely possible but exposes to a lot of subtle potential bugs.

How about checking nordpool property on first access? For instance with a property:
```diff
+    @property
+    def is_nordpool(self) -> Optional[bool]:
+        if self._nordpool is None:
+            entity_reg = er.async_get(self.hass)
+            _LOGGER.warn(entity_reg.entities.keys())
+            price_entity = entity_reg.async_get(self._price_entity)
+            if price_entity:
+                self._nordpool = price_entity.platform == "nordpool"
+            else:
+                _LOGGER.warn("Price entity is still not defined, probably a misconfiguration")
+        return self._nordpool
+
```

```diff
-        entity_reg = er.async_get(self.hass)
-        price_entity = entity_reg.async_get(self._price_entity)
-        if price_entity:
-            if price_entity.platform == "nordpool":
-                self._nordpool = True
-        else:
-            _LOGGER.warn("Price entity is not defined, configuration is probably incorrect", self._price_entity)
-
         # Process price data
         if (
-            self._nordpool
+            self.is_nordpool
             and self._price.attributes[NP_ATTR_RAW][0][NP_ATTR_START].date()
             == now.date()
         ):
```